### PR TITLE
Bnb/misc

### PR DESCRIPTION
**Pull request by bnb32**

### CODE REVIEW

File: sup3r/preprocessing/exogenous_data_handling.py [hunk 1]
```diff
@@ -102,12 +109,15 @@ def __init__(self, file_paths, features, source_h5, s_enhancements,
             if i in exo_steps:
                 for f in features:
                     if f == 'topography':
-                        data = TopoExtract(file_paths, source_h5, s_enhance,
-                                           agg_factor, target=target,
-                                           shape=shape,
-                                           raster_file=raster_file,
-                                           max_delta=max_delta,
-                                           input_handler=input_handler)
+                        print(source_file, topo_handler)
```
Comment (grantbuster) [inline sup3r/preprocessing/exogenous_data_handling.py:112]: should remove this print

File: sup3r/utilities/test_utils.py [hunk 2]
```diff
@@ -162,3 +161,69 @@ def velocity_gradient_dist(u, bins=50, range=None):
...
+    diffs = dudy - dvdx
+    diffs = diffs[(np.abs(diffs) < 14)]
+    diffs = diffs / np.sqrt(np.mean(diffs**2))
+    counts, edges = np.histogram(diffs, bins=bins, range=range)
+    centers = edges[:-1] + (np.diff(edges) / 2)
+    counts = counts.astype(float) / counts.max()
+    return centers, counts
+
+
+def ramp_rate_dist(u, v, bins=50, range=None):
```
Comment (grantbuster) [inline sup3r/utilities/test_utils.py:200]: Nice this will be super useful!

File: sup3r/utilities/topo.py [hunk 3]
```diff
@@ -242,8 +242,38 @@ class will output a topography raster corresponding to the
...
 
-        te = cls(file_paths, topo_source_h5, s_enhance, agg_factor,
+        te = cls(file_paths, topo_source, s_enhance, agg_factor,
                  target=target, shape=shape, raster_file=raster_file,
                  max_delta=max_delta, input_handler=input_handler)
 
         return te.hr_elev
+
+
+class TopoExtractNC(TopoExtract):
```
Comment (grantbuster) [inline sup3r/utilities/topo.py:252]: consider making an abstract base class with two sub classes (one for h5 and one for nc) with the two property methods over-written. Better organization and explicit subclass purposes that way.

File: sup3r/preprocessing/data_handling.py [hunk 4]
```diff
@@ -1308,8 +1308,7 @@ def run_all_data_init(self):
         logger.debug(f'Loading data for raster of shape {self.grid_shape}')
 
         # get the file-native time index without pruning
-        time_index = self.raw_time_index
-        n_steps = len(time_index[self.temporal_slice])
+        n_steps = len(self.raw_time_index[self.temporal_slice])
```
Comment (grantbuster) [inline sup3r/preprocessing/data_handling.py:1311]: people think software engineering is super complex but it's mostly creating and then fixing stupid stuff like this

**Review by grantbuster — CHANGES_REQUESTED** [review]
minor comments

File: sup3r/preprocessing/data_handling.py (hunk 4 above)
Comment (bnb32) [reply sup3r/preprocessing/data_handling.py:1311]: lol yeah

File: sup3r/utilities/topo.py (hunk 3 above)
Comment (bnb32) [reply sup3r/utilities/topo.py:252]: Yeah agreed

**Review by grantbuster — APPROVED** [review]

**Pull request merged**